CONTRIBUTING.md

```diff
@@ -0,0 +1,88 @@
+# Contributing to this repository
+
+#### Notes
+- ```master``` branch is always the 'current' branch where development for the next minor or major release is taking place
+- named branches (e.g. ```12.1```, or ```13.2```) are release branches where releases are tagged
+- All branches have gating enabled, and patches should not be merged that do not pass gating
+- QE testing is only performed on major (e.g. ```12.0.0```) and minor (e.g. ```12.1.0```) releases, not on patch (e.g. ```12.1.5```) releases, which are assumed to be adequately tested by the commit based, and periodic, jenkins gating.
+
+##### Version definitions
+
+**Major.minor.patch**
+
+* Major
+  * openstack release
+  * possibly feature?
+* minor
+  * big change within release
+  * security  - upgrade-impacting
+  * bug fixes - upgrade-impacting
+  * features
+  * sha bumps
+* patch
+  * security  - non-upgrade-impacting
+  * bug fixes - non-upgrade-impacting
+
+## Working in the repository
+
+### filing a bug report
+
+When creating an issue, please ensure you include the following information
+
+* branch (or tag) on which you encountered this issue
+* Steps to reproduce
+* Any output or logs that you gathered from your environment
+
+### fixing a bug
+
+1. A bug is filed or feature requested against ```rpc-openstack```
```

Comment:
@nrb yeah I'll leave it with numbers - I don't expect this to change too often and the numbers give the suggestion of a flow rather than a list of separate points.